internal/azcli/login.go

```diff
@@ -0,0 +1,200 @@
+// This includes automatic login detection and supports multiple authentication methods
+// including service principals, managed identities, and federated tokens.
+package azcli
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/Azure/aks-mcp/internal/command"
+	"github.com/Azure/aks-mcp/internal/config"
+)
+
+// Login types
+const (
+	AuthTypeExisting                = "existing_login"
+	AuthTypeServicePrincipal        = "service_principal"
+	AuthTypeFederatedToken          = "federated_token"
+	AuthTypeUserAssignedManagedID   = "user_assigned_managed_identity"
+	AuthTypeSystemAssignedManagedID = "system_assigned_managed_identity"
+)
+
+// validateFederatedTokenFile only allows the fixed AKS identity token path.
+func validateFederatedTokenFile(filePath string) (string, error) {
+	const allowedTokenPath = "/var/run/secrets/azure/tokens/azure-identity-token" // #nosec G101 -- not a credential, this is a fixed AKS token path
+	if filePath != allowedTokenPath {
+		return "", fmt.Errorf("federated token file path must be exactly %s", allowedTokenPath)
+	}
+	fileInfo, err := os.Stat(allowedTokenPath)
+	if err != nil {
+		return "", fmt.Errorf("cannot stat federated token file %s: %w", allowedTokenPath, err)
+	}
+	if !fileInfo.Mode().IsRegular() {
+		return "", fmt.Errorf("federated token file is not a regular file: %s", allowedTokenPath)
+	}
+	return allowedTokenPath, nil
+}
+
+// Proc is a minimal interface used by this package so tests can inject a fake process.
+type Proc interface {
+	// Run executes the given command (arguments may be included in the string) and
+	// returns the command output and an error. Implementations MUST return the
+	// combined stdout+stderr output in the string return value. The error should
+	// be non-nil when the underlying process fails to start or returns a
+	// non-zero exit status. This contract is relied upon by callers that inspect
+	// output text (for example searching for an "ERROR:" prefix) as well as
+	// the returned error value.
+	Run(cmd string) (string, error)
+}
+
+// EnsureAzCliLogin ensures az CLI is available and attempts to auto-login using environment variables
+func EnsureAzCliLogin(cfg *config.ConfigData) (string, error) {
+	if _, err := exec.LookPath("az"); err != nil {
+		return "", fmt.Errorf("az cli is not installed or not in PATH: %w", err)
+	}
+	proc := NewShellProc(cfg.Timeout)
+	return EnsureAzCliLoginWithProc(proc, cfg)
+}
+
+// NewShellProc is a package-level Proc factory used so tests can override process creation and avoid invoking the real `az` binary.
+var NewShellProc = func(timeout int) Proc {
+	return command.NewShellProcess("az", timeout)
+}
+
+// EnsureAzCliLoginWithProc is the testable implementation that uses an injected Proc.
+func EnsureAzCliLoginWithProc(proc Proc, cfg *config.ConfigData) (string, error) {
```

Comment:
The logic here differs slightly from common practice. Environment variables are typically used to override default configurations. Could we adjust the logic so that login occurs only when environment variables are set, and fallback to the default Azure configuration if none are provided?